### PR TITLE
Rename host tests to functional tests

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Rename host tests to functional tests

**Why?**
Categorize persistence integration tests and functional tests.

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by mindaugasrukas — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -93,22 +88,22 @@ COVER_ROOT                 := ./.coverage
...
-# Need the following option to have integration tests count towards coverage. godoc below:
+# Need the following option to have integration and functional tests count towards coverage. godoc below:
 # -coverpkg pkg1,pkg2,pkg3
 #   Apply coverage analysis in each test to the given list of packages.
 #   The default is for each test to analyze only the package being tested.
 #   Packages are specified as import paths.
-INTEG_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/client/...,$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/..."
-
+INTEG_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/client/...,$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/...,$(MODULE_ROOT)/temporal/..."
+FUNC_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/client/...,$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/...,$(MODULE_ROOT)/temporal/...,,$(MODULE_ROOT)/tools/..."
```
Comment (alexshtin) [inline Makefile:106]:
```suggestion
FUNCTIONAL_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/client/...,$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/...,$(MODULE_ROOT)/temporal/...,,$(MODULE_ROOT)/tools/..."
```

File: Makefile [hunk 2]
```diff
@@ -93,22 +88,22 @@ COVER_ROOT                 := ./.coverage
...
 
-# Need the following option to have integration tests count towards coverage. godoc below:
+# Need the following option to have integration and functional tests count towards coverage. godoc below:
 # -coverpkg pkg1,pkg2,pkg3
 #   Apply coverage analysis in each test to the given list of packages.
 #   The default is for each test to analyze only the package being tested.
 #   Packages are specified as import paths.
-INTEG_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/client/...,$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/..."
-
+INTEG_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/client/...,$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/...,$(MODULE_ROOT)/temporal/..."
```
Comment (alexshtin) [inline Makefile:105]: I guess this could be narrowed down to just:
```suggestion
INTEG_TEST_COVERPKG := -coverpkg="$(MODULE_ROOT)/common/persistence/..."
```

File: Makefile [hunk 3]
```diff
@@ -262,32 +257,31 @@ build-tests:
 
 unit-test: clean-test-results
 	@printf $(COLOR) "Run unit tests..."
-	@go test $(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) $(TEST_TAG) -race | tee -a test.log
+	@go test $(TEST_DIRS) -timeout=$(TEST_TIMEOUT) -tags=unittest -race | tee -a test.log
```
Comment (alexshtin) [inline Makefile:260]: Why did you add `unittest` build tag? I don't like the idea of mixing dir and tag based approaches.

File: tests/integration_test.go [hunk 4]
```diff
@@ -22,7 +22,9 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
```
Comment (alexshtin) [inline tests/integration_test.go:22]: These files need to be renamed too. You can do it later.

File: tests/ndc/test_data.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ndc
+
+import replicationpb "go.temporal.io/api/replication/v1"
+
+var (
```
Comment (alexshtin) [inline tests/ndc/test_data.go:29]: I like the idea of moving such data to `test_data.go` file!

Comment (yux0) [reply tests/ndc/test_data.go:29]: I have to do this because the build tag. When using unit test tag no go file will be built in this package and will fail.

File: Makefile (hunk 3 above)
Comment (yux0) [reply Makefile:260]: What about I change everything to use build tag?

File: Makefile (hunk 1 above)
Comment (yux0) [reply Makefile]: Updated name

File: Makefile (hunk 2 above)
Comment (yux0) [reply Makefile]: Updated

File: Makefile (hunk 3 above)
Comment (yux0) [reply Makefile]: I remove the unit test build tag and still use dir to filter tests. The problem is golang only support negative filtering. If I want to use the tag to filter integration test, I need to add `!intergrationtest` tag to all unit tests. I am not sure if there is other way around.

File: Makefile [hunk 6]
```diff
@@ -296,27 +293,23 @@ $(COVER_ROOT):
 unit-test-coverage: $(COVER_ROOT)
 	@printf $(COLOR) "Run unit tests with coverage..."
 	@echo "mode: atomic" > $(UNIT_COVER_PROFILE)
-	@go test ./$(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) -race -coverprofile=$(UNIT_COVER_PROFILE) || exit 1;
+	@go test ./$(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) -race -tags=unittest -coverprofile=$(UNIT_COVER_PROFILE) || exit 1;
```
Comment (alexshtin) [inline Makefile:296]:
```suggestion
	@go test ./$(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) -race -coverprofile=$(UNIT_COVER_PROFILE) || exit 1;
```
